src/operator/numpy/np_matrix_op-inl.h

```diff
@@ -59,6 +59,222 @@ void NumpyTranspose(const nnvm::NodeAttrs& attrs,
   }
 }
 
```

Comment:
Seems like you're re-using most of the existing [SplitOpForward](https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/matrix_op-inl.h#L2762-L2823) and [SplitOpBackward](https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/matrix_op-inl.h#L2825-L2887), maybe you could refactor the existing code to expose common SplitOpForwardImpl and SplitOpBackwardImpl and call these functions from here instead of copying the code over.